csrc/ir/nodes.cpp

```diff
@@ -2361,9 +2361,46 @@ LoadStoreOp::LoadStoreOp(
   addDataAttribute(cache_op);
 }
 
+namespace {
+// Returns the permutation from `in` to `out`, i.e., `out[perm[i]]==in[i]`.  As
+// a precondition, `out` must be a permutation of `in` per the definition of
+// std::is_permutation.
+template <typename T>
+std::vector<int64_t> Permutation(
+    const std::vector<T>& in,
+    const std::vector<T>& out) {
+  std::vector<int64_t> permutation;
+  permutation.reserve(out.size());
+  // O(n^2) is totally fine for the current use case of computing the
+  // root-to-rfactor permutation. If needed, this can be improved by requiring T
+  // to be hashable and/or comparable.
+  for (const T& out_element : out) {
+    permutation.push_back(
```

Comment:
I think what you implemented is
```C++
out[i] = in[perm[i]]
```
which does not match with what you have documented
```C++
out[perm[i]]==in[i]
```

Reply:
Good catch!